app/workers/articles/detect_animated_images_worker.rb

```diff
@@ -0,0 +1,15 @@
+module Articles
+  class DetectAnimatedImagesWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :high_priority, retry: 5, lock: :until_executing
+
+    def perform(article_id)
+      article = Article.find_by(id: article_id)
+      return unless article
+
+      detected = Articles::DetectAnimatedImages.call(article)
```

Comment:
I wonder if we should just make this another guard clause instead of an `if` at the end of the following line:

```ruby
article = Article.find_by(id: article_id)
return unless article && Articles::DetectAnimatedImages.call(article)

EdgeCache::BustArticle.call(article)
```

Reply:
I'm not a fan of side effects in boolean conditions.

I changed it to `return unless detected` to make it more explicit

Reply:
> I'm not a fan of side effects in boolean conditions.

Noted for the future. I wouldn't call it a side-effect though if the result of that operation is literally the thing you want to check in your condition. It's the main effect you are checking for, the only difference is whether or not you assign this to a variable.